### ISSUE TITLE
Breaking change in response file parsing 

### ISSUE THREAD

**Issue by matt-richardson** [dotnet/sdk#26026]
### Describe the bug

Since upgrading from SDK `6.0.300` to `6.0.301`, the response file parsing logic has changed, where

```
-p:VSTestTestAdapterPath=".;C:\BuildAgent\plugins\dotnet\tools\vstest15"
```

used to work, but now shows an error:

```
dotnet build .\IPFX.UC.PresenceServer.sln @C:\BuildAgent\temp\agentTmp\1.rsp
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.
MSBUILD : error MSB1006: Property is not valid.
Switch: C:\BuildAgent\plugins\dotnet\tools\vstest15
For switch syntax, type "MSBuild -help"
```

### To Reproduce

```
docker run -it --rm mcr.microsoft.com/dotnet/sdk:6.0.301 bash
mkdir repro
cd repro
dotnet new console
echo "-p:VSTestTestAdapterPath=\".;/opt/buildagent/plugins/dotnet/tools/vstest15\"" > params.rsp
dotnet restore repro.csproj @params.rsp
```

### Exceptions (if any)
```
MSBUILD : error MSB1006: Property is not valid.
Switch: /opt/buildagent/plugins/dotnet/tools/vstest15

For switch syntax, type "MSBuild -help"
```

### Expected
```
  Determining projects to restore...
  All projects are up-to-date for restore.
```

### Further technical details
- Include the output of `dotnet --info`
```
# dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.301
 Commit:    43f9b18481

Runtime Environment:
 OS Name:     debian
 OS Version:  11
 OS Platform: Linux
 RID:         debian.11-arm64
 Base Path:   /usr/share/dotnet/sdk/6.0.301/

Host (useful for support):
  Version: 6.0.6
  Commit:  7cca709db2

.NET SDKs installed:
  6.0.301 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.6 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.6 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version
N/A

**Comment by alvaromarithompson** [comment]
TeamCity have addressed this issue with a solution found at https://github.com/JetBrains/teamcity-dotnet-plugin/issues/158#issuecomment-1157370284

**Comment by LinardAtTenForce** [comment]
Note that this isn't only a issue with that specific parameter or with semi-colons in strings alone.

Another .rsp reproduction:

    -p:teamcity_buildConfName="Build, Test and Publish"

Error:

    Starting: .NET SDK 6.0.301 "C:\Program Files\dotnet\dotnet.exe" test [...].csproj @[...].rsp
    MSBUILD : error MSB1006: Property is not valid.
    Switch: Test and Publish
    For switch syntax, type "MSBuild -help"

**Comment by baronfel** [comment]
My suspicion is that the response file features of system.commandline are covering the MSBuild response file parsing, and so we need to do some work to disable/sidestep those features.

I haven't been able to verify that this branch contains a version of system.commandline with this feature, yet.

**Comment by loop-evgeny** [comment]
This breaks our TeamCity builds badly - same issue as [LinardAtTenForce](https://github.com/LinardAtTenForce). We have a comma in our build configuration name, so even after updating the plugin it fails due to that.

**Comment by rbhanda** [comment]
@marcpopMSFT @dsplaisted @joeloff FYI. May be they can help here

**Comment by janv8000** [comment]
https://github.com/dotnet/msbuild/issues/471 related?

**Comment by dsplaisted** [comment]
Ping @baronfel.  Does this mean we should consider reverting the behavior change instead of just working around it in the Brotli task?

**Comment by baronfel** [comment]
@janv8000 the MSBuild issue is not related.

@Danielku15 thank you for reporting a second case. @dsplaisted yes, I agree that a second case is enough to investigate reverting the S.CL update that triggered this, and then looking at how to mitigate for 6.0.400 and beyond.

**Comment by Danielku15** [comment]
I think overall there are quite some cases in the market. They have been primarily reported to JetBrains as people using TeamCity (CI server) are directly affected with this issue after upgrading the SDK 😉 Our IT rolled out .net SDK updates last night and we started suffering of these errors. 

If you need further cases you can refer to this Issue in the JetBrains issue tracker where a lot of people had the issue, but were able to patch their servers with some hotfix from JB: 
https://youtrack.jetbrains.com/issue/TW-76527/Running-a-NET-builder-task-on-DotNet-6-SDK-solution-generates-a-build-error-MSBUILD-error-MSB1006-Property-is-not-valid

**Comment by baronfel** [comment]
Closing as resolved and should be released in 6.0.303 and onwards.

**Comment by githubuser6000** [comment]
Issue is back in .400 sdk

**Comment by loop-evgeny** [comment]
Yes, easily reproducible on Linux, too, if the path contains a space:

```
~/src/with space$ dotnet new blazorwasm && dotnet publish -v detailed | grep -i "unrec"

Welcome to .NET 6.0!
---------------------
SDK Version: 6.0.400

...

The template "Blazor WebAssembly App" was created successfully.
This template contains technologies from parties other than Microsoft, see https://aka.ms/aspnetcore/6.0-third-party-notices for details.

Processing post-creation actions...
Running 'dotnet restore' on /home/em/src/with space/with space.csproj...
  Determining projects to restore...
  Restored /home/em/src/with space/with space.csproj (in 3 sec).
Restore succeeded.


         Unrecognized command or argument 'space/bin/Debug/net6.0/wwwroot/_framework/blazor.webassembly.js'.
...
```

**Comment by baronfel** [comment]
Reopening this due to repro.

**Comment by baronfel** [comment]
Closing this as we merged the servicing fix last week. This should be fixed in the October servicing release of the 6.0.300 SDK series and up.

**Comment by matt-richardson** [comment]
@baronfel - can you list the exact version(s) it's fixed in? "`6.0.300 SDK` series and up" nominally (logically?) includes `6.0.400`, which exhibits the problem.

**Comment by baronfel** [comment]
Good point, 6.0.305 and 6.0.402 should contain the fix.
